cmake/presets/all_off.cmake

```diff
@@ -0,0 +1,22 @@
+set(STANDARD_PACKAGES ASPHERE BODY CLASS2 COLLOID COMPRESS CORESHELL DIPOLE GPU
+                      GRANULAR KIM KOKKOS KSPACE LATTE MANYBODY MC MEAM MISC
+                      MOLECULE MPIIO MSCG OPT PERI POEMS
+                      PYTHON QEQ REAX REPLICA RIGID SHOCK SNAP SRD VORONOI)
+
+set(USER_PACKAGES USER-ATC USER-AWPMD USER-BOCS USER-CGDNA USER-CGSDK USER-COLVARS
+                  USER-DIFFRACTION USER-DPD USER-DRUDE USER-EFF USER-FEP USER-H5MD
+                  USER-INTEL USER-LB USER-MANIFOLD USER-MEAMC USER-MESO
+                  USER-MGPT USER-MISC USER-MOFFF USER-MOLFILE
+                  USER-NETCDF USER-OMP USER-PHONON USER-QMMM USER-QTB
+                  USER-QUIP USER-REAXC USER-SMD USER-SMTBQ USER-SPH USER-TALLY
+                  USER-UEF USER-VTK)
+
+set(PACKAGES_WITH_LIB COMPRESS GPU KIM KOKKOS LATTE MEAM MPIIO MSCG POEMS PYTHON REAX VORONOI
+                      USER-ATC USER-AWPMD USER-COLVARS USER-H5MD USER-LB USER-MOLFILE
+                      USER-NETCDF USER-QMMM USER-QUIP USER-SMD USER-VTK)
+
+set(ALL_PACKAGES ${STANDARD_PACKAGES} ${USER_PACKAGES})
+
+foreach(PKG ${ALL_PACKAGES})
+  set(PKG_${PKG} OFF CACHE BOOL "" FORCE)
```

Comment:
@junghans `all-off` should **not** be the default. Default should be a selection of packages that cover most needs of most users. That is all packages minus those that are "exotic" (e.g. USER-EFF, USER-ATC, USER-AWPMD, USER-QMMM, deprecated (MEAM, REAX), require a download or additional libraries to be installed (e.g. KIM, LATTE, MSCG, VORONOI, USER-H5MD, USER-NETCDF, USER-VTK), or additional settings/hardware (GPU, KOKKOS, USER-INTEL).

If it is easy to detect, that a serviceable local installation of an otherwise downloaded library is available (e.g. voro++, Eigen3), the default for packages that see regular use, like (VORONOI and USER-SPH that depend on those libraries) could depend on that.

If CMake is supposed to make life easier for people that have to compile LAMMPS (and keep in mind we are talking here about people that are often not familiar with compiling software or following manuals/readme file carefully), it should produce an executable compatible with most uses with just:
mkdir build; cd build; cmake ../cmake; make

Reply:
Good point, let’s keep this then!

Reply:
Good point, let’s keep this then!